File: SDKs/beezup.com/2.0/python/src/sdk/models/shared/nextcontractinfo.py
import dataclasses
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import Optional
from dataclasses_json import dataclass_json
from sdk import utils
from ..shared import fixedandvariableclickmodelinfo as shared_fixedandvariableclickmodelinfo
from ..shared import nextcontractinfolinks as shared_nextcontractinfolinks
from ..shared import variablemodelinfo as shared_variablemodelinfo


@dataclass_json
@dataclasses.dataclass
class NextContractInfo:
    r"""NextContractInfo
    Get the conditions of your next contract
    """
    
    links: shared_nextcontractinfolinks.NextContractInfoLinks = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('links') }})
    additional_click_price: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('additionalClickPrice') }})
    billing_period_in_month: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('billingPeriodInMonth') }})
    billing_period_percent_discount: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('billingPeriodPercentDiscount') }})
    click_included: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('clickIncluded') }})
    commitment_calculated_finish_utc_date: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('commitmentCalculatedFinishUtcDate'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    commitment_period_in_month: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('commitmentPeriodInMonth') }})
    contract_id: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('contractId') }})
    currency_code: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('currencyCode') }})
    discount_duration_in_month: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('discountDurationInMonth') }})
    discount_end_utc_date: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('discountEndUtcDate'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    fixed_and_variable_click_info: Optional[shared_fixedandvariableclickmodelinfo.FixedAndVariableClickModelInfo] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('fixedAndVariableClickInfo') }})
    fixed_price: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('fixedPrice') }})
    ip_user_creation: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ipUserCreation') }})
    ip_user_modification: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('ipUserModification') }})
    is_commitment_renewal_automatically: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('isCommitmentRenewalAutomatically') }})
    is_modifiable_contract: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('isModifiableContract') }})
    offer_id: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('offerId') }})
    offer_name: Optional[str] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('offerName') }})
    percent_discount: Optional[float] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('percentDiscount') }})
    start_utc_date: Optional[datetime] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('startUtcDate'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    store_count: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('storeCount') }})
    trial_period_in_month: Optional[int] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('trialPeriodInMonth') }})
    variable_model_info: Optional[shared_variablemodelinfo.VariableModelInfo] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('variableModelInfo') }})
    

File: SDKs/beezup.com/2.0/typescript/src/sdk/models/shared/nextcontractinfo.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { FixedAndVariableClickModelInfo } from "./fixedandvariableclickmodelinfo";
import { NextContractInfoLinks } from "./nextcontractinfolinks";
import { VariableModelInfo } from "./variablemodelinfo";



// NextContractInfo
/** 
 * Get the conditions of your next contract
**/
export class NextContractInfo extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=additionalClickPrice" })
  additionalClickPrice?: number;

  @SpeakeasyMetadata({ data: "json, name=billingPeriodInMonth" })
  billingPeriodInMonth?: number;

  @SpeakeasyMetadata({ data: "json, name=billingPeriodPercentDiscount" })
  billingPeriodPercentDiscount?: number;

  @SpeakeasyMetadata({ data: "json, name=clickIncluded" })
  clickIncluded?: number;

  @SpeakeasyMetadata({ data: "json, name=commitmentCalculatedFinishUtcDate" })
  commitmentCalculatedFinishUtcDate?: Date;

  @SpeakeasyMetadata({ data: "json, name=commitmentPeriodInMonth" })
  commitmentPeriodInMonth?: number;

  @SpeakeasyMetadata({ data: "json, name=contractId" })
  contractId?: string;

  @SpeakeasyMetadata({ data: "json, name=currencyCode" })
  currencyCode?: string;

  @SpeakeasyMetadata({ data: "json, name=discountDurationInMonth" })
  discountDurationInMonth?: number;

  @SpeakeasyMetadata({ data: "json, name=discountEndUtcDate" })
  discountEndUtcDate?: Date;

  @SpeakeasyMetadata({ data: "json, name=fixedAndVariableClickInfo" })
  fixedAndVariableClickInfo?: FixedAndVariableClickModelInfo;

  @SpeakeasyMetadata({ data: "json, name=fixedPrice" })
  fixedPrice?: number;

  @SpeakeasyMetadata({ data: "json, name=ipUserCreation" })
  ipUserCreation?: string;

  @SpeakeasyMetadata({ data: "json, name=ipUserModification" })
  ipUserModification?: string;

  @SpeakeasyMetadata({ data: "json, name=isCommitmentRenewalAutomatically" })
  isCommitmentRenewalAutomatically?: boolean;

  @SpeakeasyMetadata({ data: "json, name=isModifiableContract" })
  isModifiableContract?: boolean;

  @SpeakeasyMetadata({ data: "json, name=links" })
  links: NextContractInfoLinks;

  @SpeakeasyMetadata({ data: "json, name=offerId" })
  offerId?: number;

  @SpeakeasyMetadata({ data: "json, name=offerName" })
  offerName?: string;

  @SpeakeasyMetadata({ data: "json, name=percentDiscount" })
  percentDiscount?: number;

  @SpeakeasyMetadata({ data: "json, name=startUtcDate" })
  startUtcDate?: Date;

  @SpeakeasyMetadata({ data: "json, name=storeCount" })
  storeCount?: number;

  @SpeakeasyMetadata({ data: "json, name=trialPeriodInMonth" })
  trialPeriodInMonth?: number;

  @SpeakeasyMetadata({ data: "json, name=variableModelInfo" })
  variableModelInfo?: VariableModelInfo;
}
